cli/schedule_commands.go

```diff
@@ -0,0 +1,637 @@
+	}
+
+	return &schedpb.ScheduleAction{
+		Action: &schedpb.ScheduleAction_StartWorkflow{
+			StartWorkflow: newWorkflow,
+		},
+	}, nil
+}
+
+func buildScheduleState(c *cli.Context) (*schedpb.ScheduleState, error) {
+	var out schedpb.ScheduleState
+	out.Notes = c.String(FlagInitialNotes)
+	out.Paused = c.Bool(FlagInitialPaused)
+	if c.IsSet(FlagRemainingActions) {
+		out.LimitedActions = true
+		out.RemainingActions = int64(c.Int(FlagRemainingActions))
+	}
+	return &out, nil
+}
+
+func getOverlapPolicy(c *cli.Context) (enumspb.ScheduleOverlapPolicy, error) {
+	i, err := stringToEnum(c.String(FlagOverlapPolicy), enumspb.ScheduleOverlapPolicy_value)
+	if err != nil {
+		return 0, err
+	}
+	return enumspb.ScheduleOverlapPolicy(i), nil
+}
+
+func buildSchedulePolicies(c *cli.Context) (*schedpb.SchedulePolicies, error) {
+	var out schedpb.SchedulePolicies
+	var err error
+	out.OverlapPolicy, err = getOverlapPolicy(c)
+	if err != nil {
+		return nil, err
+	}
+	if c.IsSet(FlagCatchupWindow) {
+		d, err := timestamp.ParseDuration(c.String(FlagCatchupWindow))
+		if err != nil {
+			return nil, err
+		}
+		out.CatchupWindow = timestamp.DurationPtr(d)
+	}
+	out.PauseOnFailure = c.Bool(FlagPauseOnFailure)
+	return &out, nil
+}
+
+func buildSchedule(c *cli.Context) (*schedpb.Schedule, error) {
+	sched := &schedpb.Schedule{}
+	var err error
+	if sched.Spec, err = buildScheduleSpec(c); err != nil {
+		return nil, err
+	}
+	if sched.Action, err = buildScheduleAction(c); err != nil {
+		return nil, err
+	}
+	if sched.Policies, err = buildSchedulePolicies(c); err != nil {
+		return nil, err
+	}
+	if sched.State, err = buildScheduleState(c); err != nil {
+		return nil, err
+	}
+	return sched, nil
+}
+
+func getMemoAndSearchAttributesForSchedule(c *cli.Context) (*commonpb.Memo, *commonpb.SearchAttributes, error) {
+	if memoMap, err := unmarshalMemoFromCLI(c); err != nil {
+		return nil, nil, err
+	} else if memo, err := encodeMemo(memoMap); err != nil {
+		return nil, nil, err
+	} else if saMap, err := unmarshalSearchAttrFromCLI(c); err != nil {
+		return nil, nil, err
+	} else if sa, err := encodeSearchAttributes(saMap); err != nil {
+		return nil, nil, err
+	} else {
+		return memo, sa, nil
+	}
+}
+
+func CreateSchedule(c *cli.Context) error {
+	frontendClient, namespace, scheduleID, err := scheduleBaseArgs(c)
+	if err != nil {
+		return err
+	}
+	ctx, cancel := newContext(c)
+	defer cancel()
+
+	sched, err := buildSchedule(c)
+	if err != nil {
+		return err
+	}
+	memo, sa, err := getMemoAndSearchAttributesForSchedule(c)
+	if err != nil {
+		return err
+	}
+
+	req := &workflowservice.CreateScheduleRequest{
+		Namespace:        namespace,
+		ScheduleId:       scheduleID,
+		Schedule:         sched,
+		Identity:         getCliIdentity(),
+		RequestId:        uuid.New(),
+		Memo:             memo,
+		SearchAttributes: sa,
+	}
+
+	_, err = frontendClient.CreateSchedule(ctx, req)
+	if err != nil {
+		return fmt.Errorf("failed to create schedule.\n%s", err)
```

Comment:
```suggestion
		return fmt.Errorf("unable to create schedule: %s", err)
```

https://docs.temporal.io/blog/error-message-design#error-message-style-guide